server/src/main/java/org/elasticsearch/action/support/broadcast/BroadcastResponse.java

```diff
@@ -120,4 +147,23 @@ public void writeTo(StreamOutput out) throws IOException {
             exp.writeTo(out);
         }
     }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(_SHARDS_FIELD.getPreferredName());
+        builder.field(TOTAL_FIELD.getPreferredName(), totalShards);
+        builder.field(SUCCESSFUL_FIELD.getPreferredName(), successfulShards);
+        builder.field(FAILED_FIELD.getPreferredName(), failedShards);
+        if (shardFailures != EMPTY) {
+            builder.startArray(FAILURES_FIELD.getPreferredName());
+            for (DefaultShardOperationFailedException failure : shardFailures) {
+                builder.startObject();
+                failure.toXContent(builder, params);
+                builder.endObject();
+            }
+            builder.endArray();
+        }
+        builder.endObject();
+        return builder;
```

Comment:
why not calling `RestActions#buildBroadcastShardsHeader` instead ? Aren't we losing support for the `group_shard_failures` flag? It is not relevant for the high-level REST client as there is no way to set it but I think it's important given that the parsing code is in ES core. Which reminds me, we should probably test this as well in `RefreshResponseTests`. This param can be passed in as part of the `ToXContent.Params` when calling `toXContent`

Reply:
It doesn't really matter if we just call `RestActions#buildBroadcastShardsHeader` here ? seems we have talked a little before with @tlrx , like this, this method will strongly depend on `RestActions` ? Do it seperately is to not have the `XContent` building logic in the `RestRefreshAction` and "decouple" the `Response` object and REST Layer ?

Reply:
I see why we may not want to call `RestActions#buildBroadcastShardsHeader`, but I think that we are missing a part of it right now, and I am not sure we want to duplicate it. I would call it and add a TODO,  till we can move it completely to BRoadcastResponse and remove the method from RestActions. How does this sound?

Reply:
This sounds good!